entity-framework/core/what-is-new/ef-core-9.0/breaking-changes.md

```diff
@@ -56,3 +57,27 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
 ```
 
 That being said, applications should stop using sync APIs with Azure Cosmos DB since this is not supported by the Azure Cosmos DB SDK. The ability to suppress the exception will be removed in a future release of EF Core, after which the only option will be to use async APIs.
+
+## Low-impact changes
+
+<a name="sqlfunctionexpression-nullability"></a>
+
+### SqlFunctionExpression's nullability arguments' arity validated
+
+[Tracking Issue #33852](https://github.com/dotnet/efcore/issues/33852)
+
+#### Old behavior
+
+Previously it was possible to create a `SqlFunctionExpression` with a different number of arguments and nullability propagation arguments.
+
+#### New behavior
+
+Starting with EF Core 9.0, EF now throws if the number of arguments and nullability propagation arguments do not match.
+
+#### Why
+
+Not having matching number of arguments and nullability propagation arguments can lead to unexpected behavior.
+
+#### Mitigations
+
+Make sure the `argumentsPropagateNullability` has same number of elements as the `arguments`. When in doubt use `false` for nullability argument.
```

Comment:
Why the "use false when in doubt"? Can't that lead to incorrect results if the function does propagate nulls (which almost all functions do)? (BTW we don't really have to provide a tip on this at all in this breaking note)

Reply:
I'm guessing maybe I missed a design discussion on this..

Reply:
If you put `true` you might get wrong results in case you're incorrect. If you put `false` you might get less efficient query, but the result will be correct even if you're incorrect.

@maumar has opinion about including this sentence. ;)

Reply:
Discussion resolved offline.